src/main/scala/com/gu/i18n/CountryGroup.scala

```diff
@@ -312,4 +312,19 @@ object CountryGroup {
 
   def availableCurrency(currencies: Set[Currency])(country: Country) =
     byCountryCode(country.alpha2).map(_.currency).filter(currencies)
+
+  def byOptimisticCountryNameOrCode(str: String): Option[Country] = {
+    val clean = str.replace(".", "")
+    val asCode = clean.toUpperCase
+    val name = clean.toLowerCase
+
+    countryByName(name) orElse countryByCode(asCode) orElse (name match {
+      case _ if name endsWith "ireland" => Some(Country.Ireland)
```

Comment:
endsWith " of ireland"

Reply:
I don't know if we have any instances of this but it seems to me that it would cause a problem with "Northern Ireland".
I know i've seen Scotland, wales and England in the country field so I wouldn't be surprised if we had some Northern Ireland cases